src/components/Syntaxes.jsx

```diff
@@ -6,23 +6,48 @@ import Input from './common/Input';
 import * as actions from '../actions';
 
 class Syntaxes extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { activeItem: {} };
+    this.onClickHandler = this.onClickHandler.bind(this);
+  }
+
   componentDidMount() {
     const { dispatch } = this.props;
     dispatch(actions.fetchSyntaxes);
   }
 
+  onClickHandler(e) {
+    const { index, syntax } = e.target.dataset;
+
+    this.setState({ activeItem: { [index]: true } });
```

Comment:
I think I have few questions here:

- Why do we need to wrap `index` in brackets (i.e. `[index]`)? What does it mean? Why just not `index`?
- Why does `activeItem` is an object with just one `bool` which can't be even `false`? I'd suggest to use the following schema:
  ```javascript
  this.setState({ activeIndex: index })
  ```
  What do you think?

Reply:
1. look on one of answers above about these [] brackets;
2. and 3. Take a look on line 39 where I check is newly selected item has appropriate index, in this way I can avoid additional loops that will do nothing more than remove active class from previous item